mesonbuild/modules/python.py

```diff
@@ -64,29 +64,46 @@ def __init__(self, python_holder, environment, kwargs):
         else:
             self.major_version = 2
 
+        # 1. We first try to find the necessary python
+        #    variables using pkg-config...
         if DependencyMethods.PKGCONFIG in self.methods and not python_holder.is_pypy:
             pkg_version = self.variables.get('LDVERSION') or self.version
             pkg_libdir = self.variables.get('LIBPC')
-            old_pkg_libdir = os.environ.get('PKG_CONFIG_LIBDIR')
-            old_pkg_path = os.environ.get('PKG_CONFIG_PATH')
 
-            os.environ.pop('PKG_CONFIG_PATH', None)
+            # a. If python-X.Y.pc exists in LIBPC, we will try to use it
+            if Path(os.path.join(pkg_libdir, 'python-{}.pc'.format(pkg_version))).is_file():
+                old_pkg_libdir = os.environ.get('PKG_CONFIG_LIBDIR')
+                old_pkg_path = os.environ.get('PKG_CONFIG_PATH')
 
-            if pkg_libdir:
-                os.environ['PKG_CONFIG_LIBDIR'] = pkg_libdir
+                os.environ.pop('PKG_CONFIG_PATH', None)
 
-            try:
-                self.pkgdep = PkgConfigDependency('python-{}'.format(pkg_version), environment, kwargs)
-            except Exception:
-                pass
+                if pkg_libdir:
+                    os.environ['PKG_CONFIG_LIBDIR'] = pkg_libdir
 
-            if old_pkg_path is not None:
-                os.environ['PKG_CONFIG_PATH'] = old_pkg_path
+                try:
+                    self.pkgdep = PkgConfigDependency('python-{}'.format(pkg_version), environment, kwargs)
+                    mlog.debug('Found "python-{}" via pkg-config lookup in LIBPC ({})'.format(pkg_version, pkg_libdir))
+                except Exception:
```

Comment:
This should be `MesonException`. The original code used `Exception`, but that's wrong too.

Also we should log the exception using mlog.debug.